File: 101-SymmetricTree/101-SymmetricTree.java
// Last updated: 09/07/2026, 10:09:57
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
        return ismirror(root.left,root.right);
    }
    boolean ismirror(TreeNode left,TreeNode right){
        if(left == null && right == null) return true;
        if(left == null || right == null) return false;
        if(left.val!=right.val) return false;

        boolean outer = ismirror(left.left,right.right);
        boolean inner = ismirror(left.right,right.left);
        return outer && inner;
    }
}

File: 101-SymmetricTree/101-SymmetricTree.py
# Last updated: 09/07/2026, 10:10:02
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        return self.check(root.left,root.right)
        
    def check(self,left,right):
        if left is None and right is None:
            return True
        if left is None or right is None:
            return False
        
        if left.val != right.val:
            return False    
        return self.check(left.left, right.right) and self.check(left.right, right.left)           